docs/site/Middleware.md

```diff
@@ -0,0 +1,282 @@
+      const result = await this.invoke(route, args);
+
+      debug('%s result -', route.describe(), result);
+      this.send(response, result);
+    } catch (error) {
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+#### Extend sequence with more than one middleware actions
+
+Sometimes we want to add middleware to the sequence in between other actions,
+for example, do some post-processing before the result is written to the HTTP
+response. This can be achieved by overriding the sequence implementation to call
+`invokeMiddleware` as needed.
+
+{% include code-caption.html content="src/sequence.ts" %}
+
+```ts
+export class SequenceWithMiddleware extends DefaultSequence {
+  async handle(context: RequestContext): Promise<void> {
+    try {
+      const {request, response} = context;
+      // The default middleware chain
+      await this.invokeMiddleware(context);
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+
+      // The second middleware chain for post-invocation processing
+      context.bind('invocation.result').to(result);
+      // POST_INVOCATION_MIDDLEWARE is the name of the extension point
+      // for post-invocation middleware
+      await this.invokeMiddleware(context, {
+        extensionPoint: POST_INVOCATION_MIDDLEWARE,
+      });
+      this.send(response, result);
+    } catch (error) {
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+It's also possible to inject multiple instances of `InvokeMiddleware`, each of
+which has its own extension point name.
+
+{% include code-caption.html content="src/sequence.ts" %}
+
+```ts
+export class SequenceWithMiddleware extends DefaultSequence {
+  /**
+   * Optional middleware chain
+   * Invokes registered middleware (injected via SequenceActions.MIDDLEWARE).
+   */
+  @inject('middleware.postInvoke', {optional: true})
+  protected middlewarePostInvoke: InvokeMiddleware = () => {};
+
+  async handle(context: RequestContext): Promise<void> {
+    try {
+      const {request, response} = context;
+      // The default middleware chain
+      await this.invokeMiddleware(context);
+      const route = this.findRoute(request);
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+
+      // The second middleware chain for post-invocation processing
+      context.bind('invocation.result').to(result);
+      await this.middlewarePostInvoke(context);
+      this.send(response, result);
+    } catch (error) {
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+Now we can set up applications to leverage the new sequence:
+
+{% include code-caption.html content="src/application.ts" %}
+
+```ts
+export class MyApplication extends RestApplication {
+  constructor(config: ApplicationConfig) {
+    // Create another middleware phase. This is NOT needed if we use the same
```

Comment:
Not sure what the comment means. Can you clarify?  we seem to be using invokeMiddleware WITH a different extension, so the comment is confusing.